connect-inject/metrics_configuration.go

```diff
@@ -0,0 +1,194 @@
+package connectinject
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type MetricsConfiguration struct {
+	DefaultEnableMetrics        bool
+	DefaultEnableMetricsMerging bool
+	DefaultMergedMetricsPort    string
+	DefaultPrometheusScrapePort string
+	DefaultPrometheusScrapePath string
+}
+
+const (
+	defaultServiceMetricsPath = "/metrics"
+)
+
+// mergedMetricsServerConfiguration is called when running a merged metrics server and used to return ports necessary to
+// configure the merged metrics server.
+func (mc MetricsConfiguration) mergedMetricsServerConfiguration(pod corev1.Pod) (string, string, string, error) {
+	run, err := mc.shouldRunMergedMetricsServer(pod)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// This should never happen because we only call this function in the handler if
+	// we need to run the metrics merging server. This check is here just in case.
+	if !run {
+		return "", "", "", errors.New("metrics merging should be enabled in order to return the metrics server configuration")
+	}
+
+	// Configure consul sidecar with the appropriate metrics flags.
+	mergedMetricsPort, err := mc.mergedMetricsPort(pod)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// Don't need to check the error since it's checked in the call to
+	// mc.shouldRunMergedMetricsServer() above.
+	serviceMetricsPort, _ := mc.serviceMetricsPort(pod)
+
+	serviceMetricsPath := mc.serviceMetricsPath(pod)
+
+	return mergedMetricsPort, serviceMetricsPort, serviceMetricsPath, nil
+}
+
+// enableMetrics returns the default value in the handler, or overrides that
```

Comment:
super minor nit: but this function and the next have the same comment, maybe their comments could be changed so they explain how they're not the same!

Reply:
yeah that's a good point, I updated this!